### PR TITLE
Added host multithreading support for FFTW

### PR DESCRIPTION
**Pull request by aayushg55**
- Enabled usage of multi-threaded FFTW
- Templated host executor for single or multi-threaded operation
- Added multithreading to host Exec() call
- Resolved host FFT bugs when a temp output tensor is required

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -171,13 +172,30 @@ set(WARN_FLAGS ${WARN_FLAGS} $<$<COMPILE_LANGUAGE:CXX>:-Werror>)
...
-    #find_package(nvpl REQUIRED COMPONENTS fft)
-    #target_link_libraries(matx INTERFACE nvpl::fftw)
-    target_link_libraries(matx INTERFACE nvpl_fftw)
-    target_compile_definitions(matx INTERFACE MATX_EN_NVPL=1)
+# Host support
+if (MATX_EN_NVPL OR MATX_EN_X86_FFTW)
+    message(STATUS "Enabling OpenMP support")
+    find_package(OpenMP REQUIRED)
+    target_link_libraries(matx INTERFACE OpenMP::OpenMP_CXX)
+    target_link_libraries(matx INTERFACE )
```
Comment (cliffburdick) [inline CMakeLists.txt:180]: This looks like a typo

File: CMakeLists.txt [hunk 2]
```diff
@@ -171,13 +172,30 @@ set(WARN_FLAGS ${WARN_FLAGS} $<$<COMPILE_LANGUAGE:CXX>:-Werror>)
...
+        target_link_libraries(matx INTERFACE nvpl::fftw)
+        target_compile_definitions(matx INTERFACE MATX_EN_NVPL=1)
+    else()
+        if (MATX_EN_X86_FFTW)
+            message(STATUS "Enabling x86 FFTW")
+            find_library(FFTW_LIB fftw3 REQUIRED)
+            find_library(FFTWF_LIB fftw3f REQUIRED)
+            find_library(FFTW_OMP_LIB fftw3_omp REQUIRED)
+            find_library(FFTWF_OMP_LIB fftw3f_omp REQUIRED)
+            list(APPEND FFTW_LIBS ${FFTW_LIB} ${FFTWF_LIB} ${FFTW_OMP_LIB} ${FFTWF_OMP_LIB})
```
Comment (cliffburdick) [inline CMakeLists.txt:194]: Is this needed or can you just put it on the next line since it's only used once?

File: CMakeLists.txt (hunk 1 above)
Comment (aayushg55) [reply CMakeLists.txt:180]: Yes, will delete the second target_link_libraries

File: CMakeLists.txt (hunk 2 above)
Comment (aayushg55) [reply CMakeLists.txt:194]: I can remove the intermediate list

**Review by cliffburdick — APPROVED** [review]

**Pull request merged**